.github/actions/test_deploy_ubuntu.sh

```diff
@@ -0,0 +1,241 @@
+#!/bin/bash
+
+# Copyright (C) 2015, Wazuh Inc.
+# March 6, 2019.
+#
+# This program is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public
+# License (version 2) as published by the FSF - Free Software
+# Foundation.
+
+# Global variables
+VERSION="$(cat src/VERSION | sed 's/v//')"
+MAJOR=$(echo ${VERSION} | cut -dv -f2 | cut -d. -f1)
+MINOR=$(echo ${VERSION} | cut -d. -f2)
+SHA="$(git rev-parse --short $1)"
+
+
+WAZUH_MANAGER="1.1.1.1"
+WAZUH_MANAGER_PORT="7777"
+WAZUH_PROTOCOL="udp"
+WAZUH_REGISTRATION_SERVER="2.2.2.2"
+WAZUH_REGISTRATION_PORT="8888"
+WAZUH_REGISTRATION_PASSWORD="password"
+WAZUH_KEEP_ALIVE_INTERVAL="10"
+WAZUH_TIME_RECONNECT="10"
+WAZUH_REGISTRATION_CA="/var/ossec/etc/testsslmanager.cert"
+WAZUH_REGISTRATION_CERTIFICATE="/var/ossec/etc/testsslmanager.cert"
+WAZUH_REGISTRATION_KEY="/var/ossec/etc/testsslmanager.key"
+WAZUH_AGENT_NAME="test-agent"
+WAZUH_AGENT_GROUP="test-group"
+ENROLLMENT_DELAY="10"
+
+WAZUH_REGISTRATION_PASSWORD_PATH="etc/authd.pass"
+    
+function install_wazuh(){
+  echo "Testing the following variables $@"
+  eval "${@} apt install -y ./wazuh-agent_${VERSION}-0.commit${SHA}_amd64.deb > /dev/null 2>&1"
+}
+
+function remove_wazuh () {
+  apt purge -y wazuh-agent > /dev/null 2>&1
+}
+
+function test() {
```

Comment:
I see too much-repeated code. Please implement a template function (or functions) that avoids the code repeating.